### ISSUE TITLE
feature: hasSuperMethod or isSuperMethodOverride

### ISSUE THREAD

**Issue by trusktr** [ajafff/tsutils#129]
This would return `true` if the given class method overrides a super class method (has a super method).

```js
function hasSuperMethod(method: ts.SyntaxKind.MethodDeclaration): boolean
// or something. I'm new to this.
```

This could be useful, for example, in making a typescript-eslint rule that can throw a warning if a subclass method body forgets to call a super method.

Forgetting to call a super method is all too easy, and sometimes the runtime effects of doing so can go unnoticed for a while before unexpected behavior happens.

**Comment by ajafff** [comment]
This makes sense as a utility. I have already written such code in a few places, it's about time to make it reusable.

I would probably widen the scope to something like
```ts
function getBaseClassMemberOfClassElement(node: ts.PropertyDeclaration | ts.MethodDeclaration | ts.AccessorDeclaration, checker: ts.TypeChecker): ts.Symbol | undefined
```

That means is works on other class elements as well:
```ts
class Base {
    method() {}
}

class Derived extends Base {
    method = () => {
        // forgot super.method()?
    }
}
```

Instead of returning a boolean it returns a `ts.Symbol` (if the member exists in any base class). By looking at `ts.Symbol#declarations` you can determine how the member was declared in the base class: static or abstract property, method or accessor. You could even look at the JSDoc of the base class member, e.g. if you want to activate your lint rule with a JSDoc tag like `/** @mustCallSuper */`.

Of course it should handle static members and members with computed property name as expected.

Note to myself: private named instance fields are not inherited/overridden. Therefore it should **not** return a private named instance field of the base class if one exists with the same name.

**Comment by ajafff** [comment]
This is now implemented as described above and released in `v3.19.0`
